### PR TITLE
DLS-10416 Check your details at the start

### PR DESCRIPTION
**Pull request by johno1985**

### CODE REVIEW

**Comment by mahesh-hmrc** [comment]
Quick check and noticed that we don't have test coverage added for the new controller `CheckYourDetailsController`. If we can add that. Will go through rest of the PR. Thanks

File: app/uk/gov/hmrc/helptosavefrontend/controllers/EligibilityCheckController.scala [hunk 1]
```diff
@@ -184,7 +184,11 @@ class EligibilityCheckController @Inject() (
   def getMissingInfoPage: Action[AnyContent] =
     authorisedForHtsWithInfo { implicit request => implicit htsContext =>
       htsContext.userDetails.fold(
-        missingInfo => Ok(missingUserInfo(missingInfo.missingInfo)),
+        { missingInfo =>
+          // DLS-10416 - Check in Kibana to see whether we can remove the Date of Birth warning
+          logger.info("Missing user info: " + missingInfo.missingInfo.toString())
```
Comment (mahesh-hmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/EligibilityCheckController.scala:189]: Same as above, would be ideal not to log all details, but the fact that DOB is missing

File: test/uk/gov/hmrc/helptosavefrontend/controllers/CheckYourDetailsControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        contentAsString(result).trim should include(
+          "Your details"
+        )
+
+      }
+    }
+  }
+
+  private def doCommonChecks(doRequest: () => Future[Result]): Unit = {
+//    "redirect user to eligibility checks if there is no session found" in {
```
Comment (mahesh-hmrc) [inline test/uk/gov/hmrc/helptosavefrontend/controllers/CheckYourDetailsControllerSpec.scala:70]: Is this still in works? Call to this is made in above case. if not required, can remove

File: app/uk/gov/hmrc/helptosavefrontend/controllers/CheckYourDetailsController.scala [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  ec: ExecutionContext
+) extends BaseController(cpd, mcc, errorHandler, maintenanceSchedule) with HelpToSaveAuth with EnrolmentCheckBehaviour
+    with Logging {
+
+  def checkYourDetails: Action[AnyContent] =
+    authorisedForHtsWithInfo { implicit request => implicit htsContext =>
+      checkIfAlreadyEnrolled { () =>
+        htsContext.userDetails match {
+          case Left(missingUserInfos) =>
+            logger.warn(s"user data missing: $missingUserInfos")
```
Comment (mahesh-hmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/CheckYourDetailsController.scala:55]: Probably ideal not to log the whole user info here. Has full name, DOB and contact details.
I understand this is to track the volume of users missing the DOB field, mentioned in DLS-10416.

However, if it's for that we are logging, better would be to just log the fact DOB is missing than the actual details.

File: test/uk/gov/hmrc/helptosavefrontend/controllers/CheckYourDetailsControllerSpec.scala (hunk 2 above)
Comment (johno1985) [reply test/uk/gov/hmrc/helptosavefrontend/controllers/CheckYourDetailsControllerSpec.scala:70]: Yep, still WIP, apologies, I should have marked it as such

File: app/uk/gov/hmrc/helptosavefrontend/controllers/CheckYourDetailsController.scala (hunk 3 above)
Comment (johno1985) [reply app/uk/gov/hmrc/helptosavefrontend/controllers/CheckYourDetailsController.scala:55]: I believe this only logs field names, not the actual values

Comment (mahesh-hmrc) [reply app/uk/gov/hmrc/helptosavefrontend/controllers/CheckYourDetailsController.scala:55]: Oh wow, just had a look, it's a separate case class altogether for this, thought it was coming off userinfo. pretty sure that could be lot more simplified by data validation (similar to cats data) on the original userinfo model and log the fields missing

Comment (johno1985) [reply app/uk/gov/hmrc/helptosavefrontend/controllers/CheckYourDetailsController.scala:55]: Probably, but I didn't want to go and re-write everything that Lukasz did

File: app/uk/gov/hmrc/helptosavefrontend/controllers/EligibilityCheckController.scala (hunk 1 above)
Comment (mahesh-hmrc) [reply app/uk/gov/hmrc/helptosavefrontend/controllers/EligibilityCheckController.scala:189]: Discussion in the other message, nothing to address now

**Review by mahesh-hmrc — APPROVED** [review]

**Pull request merged**